crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -218,7 +221,23 @@ impl SargonOS {
             .radix_engine_toolkit_receipt
             .ok_or(CommonError::FailedToExtractTransactionReceiptBytes)?;
 
-        manifest.execution_summary(engine_toolkit_receipt)
+        let execution_summary =
+            manifest.execution_summary(engine_toolkit_receipt)?;
+
+        let forbidden_manifest_class = execution_summary
+            .detailed_classification
+            .iter()
+            .find(|classification| classification.is_forbidden());
+
+        if let Some(forbidden_manifest_class) = forbidden_manifest_class
+            && !are_instructions_originating_from_host
+        {
+            return Err(CommonError::ForbiddenManifestClass {
+                class: forbidden_manifest_class.clone(),
+            });
+        }
```

Comment:
Note:
Currently, for delete account this is paranoia check. There are no tests for it, as it cannot happen for the following reasons:
- AccountSecurify is a reserved instruction, so it our current delete manifest can be submitted only by the Wallet, and is pre-validated when we do check for reserved instructions.
- Then, the other way when a manifest can be classified as Delete Accounts, is doing a transfer of the badge, which is not possible - the access controller does not allow withdrawing a non-fungible and transferring it, and we cannot withdraw the badge from a deleted account because it is tombstoned.